File: 240206/G or H 3/G-or-H-3.py
n,k=map(int,input().split())

placed=[0]*10001

for _ in range(n):
    num,sign=input().split()
    if sign=='G':
        placed[int(num)]=1
    else:
        placed[int(num)]=2


ans=0
for i in range(1,10000-k+1):
    val=0
    for j in range(i,i+k+1):
        val+=placed[j]
    ans=max(val,ans)

print(ans)

File: 260110/G or H 3/G-or-H-3.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");
const [n, k] = input[0].split(' ').map(Number);
const people = input.slice(1).map((m)=>m.split(' '));

const arr = Array(20000+1).fill(0);

for(let i=0; i<people.length; i++){
  const [idx, c] = people[i];
  arr[Number(idx)] = c==='G'?1:2
}

let ans = 0;
for(let i = 1; i<20001-k; i++){
  let sum = 0;
  for(let j=i; j<=i+k; j++){
    sum+=arr[j];
  }
  ans = Math.max(ans, sum);
}

console.log(ans)


